From 80b375f1ff3822209831d783600a6ef6c7bba230 Mon Sep 17 00:00:00 2001
From: Marius Vlad <marius.vlad@collabora.com>
Date: Tue, 23 Mar 2021 09:57:52 +0200
Subject: [PATCH 3/7] kiosk-shell: Enable screenshooter in kiosk-shell

Signed-off-by: Marius Vlad <marius.vlad@collabora.com>
---
 kiosk-shell/kiosk-shell.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/kiosk-shell/kiosk-shell.c b/kiosk-shell/kiosk-shell.c
index 02d1e152..5f353182 100644
--- a/kiosk-shell/kiosk-shell.c
+++ b/kiosk-shell/kiosk-shell.c
@@ -1114,6 +1114,8 @@ wet_shell_init(struct weston_compositor *ec,
 	shell->output_moved_listener.notify = kiosk_shell_handle_output_moved;
 	wl_signal_add(&ec->output_moved_signal, &shell->output_moved_listener);
 
+	screenshooter_create(ec);
+
 	kiosk_shell_add_bindings(shell);
 
 	return 0;
-- 
2.29.2

